llvm/lib/ObjectYAML/DXContainerEmitter.cpp

```diff
@@ -276,10 +276,6 @@ void DXContainerWriter::writeParts(raw_ostream &OS) {
       for (DXContainerYAML::RootParameterLocationYaml &L :
            P.RootSignature->Parameters.Locations) {
 
-        assert(dxbc::isValidParameterType(L.Header.Type) &&
-               "invalid DXContainer YAML");
-        assert(dxbc::isValidShaderVisibility(L.Header.Visibility) &&
-               "invalid DXContainer YAML");
         dxbc::RootParameterType Type = dxbc::RootParameterType(L.Header.Type);
         dxbc::ShaderVisibility Visibility =
             dxbc::ShaderVisibility(L.Header.Visibility);
```

Comment:
The casts are no longer needed, since these are already the right type now. Here and throughout this file.